cardinal/ecs/persona_test.go

```diff
@@ -114,3 +115,34 @@ func TestDuplicatePersonaTagsInTickAreOnlyRegisteredOnce(t *testing.T) {
 	// The saved address should be unchanged
 	assert.Equal(t, addr, "address-0")
 }
+
+func TestCanAuthorizeAddress(t *testing.T) {
+	// Verify that the CreatePersonaTransaction is automatically created and registered with a world.
+	world := inmem.NewECSWorldForTest(t)
+	assert.NilError(t, world.LoadGameState())
+
+	wantTag := "CoolMage"
+	wantSigner := "123-456"
+	ecs.CreatePersonaTx.AddToQueue(world, ecs.CreatePersonaTransaction{
+		PersonaTag:    wantTag,
+		SignerAddress: wantSigner,
+	})
+
+	wantAddr := "0xfoobar"
+	ecs.AuthorizePersonaAddressTx.AddToQueue(world, ecs.AuthorizePersonaAddress{
+		PersonaTag: wantTag,
+		Address:    wantAddr,
+	}, &sign.SignedPayload{PersonaTag: wantTag})
+	// PersonaTag registration doesn't take place until the relevant system is run during a game tick.
+	assert.NilError(t, world.Tick(context.Background()))
+
+	ecs.NewQuery(filter.Exact(ecs.SignerComp)).Each(world, func(id storage.EntityID) bool {
```

Comment:
nit: add a count variable before this ecs.NewQuery line, increment the count by 1 in the Each function, and verify the count was incremented. If this NewQuery line fails to find any matches, nonce of the asserts will be hit and it will look like a passing test.

Reply:
ah right. good catch. fixed